lib/proxy/helpers.rb

```diff
@@ -20,7 +20,7 @@ def log_halt(code = nil, exception_or_msg = nil, custom_msg = nil)
     rescue => e
       exception = e
       message += e.message
-      code ||= 400
+      code ||= e.status_code
```

Comment:
Please revert this. It's wrong to change the API this way since not every exception has a status code, which in turn will cause internal server errors:
```ruby
[1] pry(main)> IndexError.status_code
NoMethodError: undefined method `status_code' for IndexError:Class
from (pry):2:in `__pry__'
```

Reply:
Fixed in https://github.com/theforeman/smart-proxy/pull/869